server/modules/survey/manager/surveyManager.js

```diff
@@ -271,17 +272,24 @@ export const publishSurveyProps = async (surveyId, langsDeleted, client = db) =>
     }
   })
 
-export const { updateSurveyDependencyGraphs } = SurveyRepository
+export const { removeSurveyTemporaryFlag, updateSurveyDependencyGraphs } = SurveyRepository
 
 // ====== DELETE
-export const deleteSurvey = async (surveyId) =>
-  db.tx(async (t) => {
-    await Promise.all([
-      UserRepository.deleteUsersPrefsSurvey(surveyId, t),
-      SurveyRepository.dropSurveySchema(surveyId, t),
-      SchemaRdbRepository.dropSchema(surveyId, t),
-      SurveyRepository.deleteSurvey(surveyId, t),
-    ])
+export const deleteSurvey = async (surveyId, { deleteUserPrefs = true } = {}, client = db) =>
+  client.tx(async (t) => {
+    if (deleteUserPrefs) {
+      await UserRepository.deleteUsersPrefsSurvey(surveyId, t)
+    }
+    await SurveyRepository.deleteSurvey(surveyId, t)
+    await SurveyRepository.dropSurveySchema(surveyId, t)
+    await SchemaRdbRepository.dropSchema(surveyId, t)
+  })
+
+export const deleteTemporarySurveys = async ({ olderThan24Hours }, client = db) =>
+  client.tx(async (t) => {
+    const surveyIds = await SurveyRepository.fetchTemporarySurveyIds({ olderThan24Hours }, t)
+    await PromiseUtils.each(surveyIds, async (surveyId) => deleteSurvey(surveyId, { deleteUserPrefs: true }, t))
```

Comment:
maybe this can be done in parallel but maybe is safer to do It sequential

Reply:
there will be few stale temporary surveys (ideally no one); doing it sequentially makes the logging nicer... otherwise in case of problems is difficult to find the survey causing the problem

Reply:
okii